orc/src/main/java/org/apache/iceberg/orc/OrcIterable.java

```diff
@@ -59,10 +59,17 @@
   @SuppressWarnings("unchecked")
   @Override
   public Iterator<T> iterator() {
+    final Reader orcFileReader = newFileReader(file, config);
+    final ColumnMap colMapping = orcFileReader.hasMetadataValue(ORC.COLUMN_NUMBERS_ATTRIBUTE) ?
+        ColumnMap.deserialize(orcFileReader.getSchema(),
+            orcFileReader.getMetadataValue(ORC.COLUMN_NUMBERS_ATTRIBUTE)) : new ColumnMap();
+    // Get the ORC schema to read based on the Icebert to ORC mapping
+    final TypeDescription readOrcSchema = TypeConversion.toOrc(schema, colMapping);
```

Comment:
It seems currently, colMapping will always reflect the current Iceberg reader schema since it is being updated in `TypeConversion` class. We'd need a TODO or ticket here to address schema evolution by column Id correctly

Reply:
Added `TODO`.